spot-contracts/contracts/RolloverVault.sol

```diff
@@ -542,6 +661,56 @@ contract RolloverVault is
         }
     }
 
+    /// @dev Tranches the vault's underlying to mint perps as needed.
+    ///      If the vault already holds required perps, it skips minting new ones.
+    ///      Additionally, performs some book-keeping to keep track of the vault's assets.
+    function _trancheAndMintPerps(
+        IPerpetualTranche perp_,
+        uint256 perpPrice,
+        uint256 perpAmtToMint
+    ) private {
+        // If the vault already has a perp bal, mint only the delta
+        uint256 perpBal = perp.balanceOf(address(this));
+        perpAmtToMint = (perpAmtToMint > perpBal) ? perpBal - perpAmtToMint : 0;
+
+        // Skip if mint amount is zero
+        if (perpAmtToMint <= 0) {
+            return;
+        }
+
+        // Get the minting bond data
+        IBondController depositBond = perp.getDepositBond();
+        BondTranches memory bt = depositBond.getTranches();
+
+        // NOTE: we use only the most senior tranche to mint perps
+        ITranche tranche = bt.tranches[0];
+
+        // Compute the senior tranche amount need to be deposited in order to mint perps.
+        uint256 seniorAmtToDeposit = perpAmtToMint.mulDiv(
+            perpPrice,
```

Comment:
perp.getAvgPrice() -> returns the balance weighted price of all the tranches in the reserve
perp.computePrice(tranche) -> returns the price of a given tranche

Reply:
But here we could use the vault's `_computeVaultTrancheValue` method rather than `perp.computePrice`, if we are okay assuming both contracts implement the same pricing strategy. However, Asking perp for how the tranche is priced is guaranteed to be correct.

Reply:
This should be moot, if we move to the design with different mint functions on Perp

Reply:
will plan to move this logic to preview deposit amount into perp